oss-internship-2020/openjpeg/examples/decompress_example.cc

```diff
@@ -0,0 +1,159 @@
+        .AllowOpen()
+        .AllowRead()
+        .AllowWrite()
+        .AllowStat()
+        .AllowSystemMalloc()
+        .AllowExit()
+        .AllowSyscalls({
+            __NR_futex,
+            __NR_close,
+            __NR_lseek,
+        })
+        .AddFile(in_file_)
+        .BuildOrDie();
+  }
+
+ private:
+  std::string in_file_;
+};
+
+int main(int argc, char* argv[]) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+
+  if (argc != 3) {
+    std::cerr << "usage: " << basename(argv[0]) << " absolute/path/to/INPUT.jp2"
+              << " absolute/path/to/OUTPUT.pnm\n";
+    return EXIT_FAILURE;
+  }
+
+  std::string in_file(argv[1]);
+
+  // initialize sandbox
+  Openjp2SapiSandbox sandbox(in_file);
+  absl::Status status = sandbox.Init();
+  assert(status.ok());
+
+  Openjp2Api api(&sandbox);
+  sapi::v::ConstCStr in_file_v(in_file.c_str());
+
+  // initialize library's main data-holders
+  sapi::StatusOr<opj_stream_t*> stream_status =
+      api.opj_stream_create_default_file_stream(in_file_v.PtrBefore(), 1);
+  assert(stream_status.ok());
+  void* stream_status_value = stream_status.value();
+  sapi::v::RemotePtr stream_pointer(stream_status_value);
+
+  sapi::StatusOr<opj_codec_t*> codec_status =
+      api.opj_create_decompress(OPJ_CODEC_JP2);
+  assert(codec_status.ok());
+  void* codec_status_value = codec_status.value();
+  sapi::v::RemotePtr codec_pointer(codec_status_value);
+
+  Parameters parameters;
+  status = api.opj_set_default_decoder_parameters(parameters.PtrBoth());
+  assert(status.ok());
+
+  sapi::StatusOr<OPJ_BOOL> bool_status =
+      api.opj_setup_decoder(&codec_pointer, parameters.PtrBefore());
+  assert(bool_status.ok());
+  assert(bool_status.value());
+
+  // start reading image from the input file
+  sapi::v::GenericPtr image_pointer;
+  bool_status = api.opj_read_header(&stream_pointer, &codec_pointer,
+                                    image_pointer.PtrAfter());
+  assert(bool_status.ok());
+  assert(bool_status.value());
+
+  Opj_image_t image;
+  image.SetRemote((void*)image_pointer.GetValue());
+  assert(sandbox.TransferFromSandboxee(&image).ok());
+
+  bool_status = api.opj_decode(&codec_pointer, &stream_pointer,
+                               (sapi::v::Ptr*)&image_pointer);
+  assert(bool_status.ok());
+  assert(bool_status.value());
+
+  bool_status = api.opj_end_decompress(&codec_pointer, &stream_pointer);
+  assert(bool_status.ok());
+  assert(bool_status.value());
+
+  int components = image.data().numcomps;
+
+  // transfer the read data to the main process
+  sapi::v::Array<opj_image_comp_t> image_components(components);
+  image_components.SetRemote(image.data().comps);
+  assert(sandbox.TransferFromSandboxee(&image_components).ok());
+
+  image.mutable_data()->comps = (opj_image_comp_t*)image_components.GetLocal();
+
+  int width = (int)image.data().comps[0].w;
+  int height = (int)image.data().comps[0].h;
+
+  OPJ_INT32 data[components][width * height];
```

Comment:
Variable length arrays are not part of standard C++
https://google.github.io/styleguide/cppguide.html#Nonstandard_Extensions